### PR TITLE
Allow custom webpack config

### PR DESCRIPTION
**Pull request by soupette**
#### Description of what you did:
This PR allows to modify the webpack configuration

### CODE REVIEW

File: packages/strapi-admin/index.js [hunk 1]
```diff
@@ -9,6 +10,64 @@ const chokidar = require('chokidar');
 
 const getPkgPath = name => path.dirname(require.resolve(`${name}/package.json`));
 
+function getCustomWebpackConfig(dir, config) {
+  const adminConfigPath = path.join(dir, 'admin', 'admin.config.js');
+  let webpackConfig = _.cloneDeep(getWebpackConfig(config));
```
Comment (alexandrebodin) [inline packages/strapi-admin/index.js:15]: deep cloning the webpackconfig might create unintended bugs as it is not just a plain js object but also contains some plugin instance an such.

File: packages/strapi-admin/index.js [hunk 2]
```diff
@@ -9,6 +10,64 @@ const chokidar = require('chokidar');
...
+
+  if (fs.existsSync(adminConfigPath)) {
+    const adminConfig = require(path.resolve(adminConfigPath));
+
+    if (_.isFunction(adminConfig.webpack)) {
+      webpackConfig = adminConfig.webpack(webpackConfig, webpack);
```
Comment (alexandrebodin) [inline packages/strapi-admin/index.js:21]: You should throw an error if the call returns undefined and add a nice log message

File: packages/strapi-admin/index.js (hunk 1 above)
Comment (soupette) [reply packages/strapi-admin/index.js:15]: So you suggest I don't clone it letting the user mutate it directly?

File: packages/strapi-admin/index.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-admin/index.js:21]: Yes absolutely!

File: packages/strapi-admin/index.js (hunk 1 above)
Comment (alexandrebodin) [reply packages/strapi-admin/index.js:15]: yes I think it is why other are doing it like this.

**Comment by alexandrebodin** [comment]
Will be really cool for the users

**Pull request merged**